src/openrct2/world/Climate.cpp

```diff
@@ -223,7 +223,7 @@ void ClimateForceWeather(WeatherType weather)
     gameState.ClimateCurrent.Temperature = pattern.baseTemperature + trait.temperatureDelta;
     gameState.ClimateUpdateTimer = 1920;
 
-    ClimateUpdate();
+    ClimateDetermineFutureWeather(ScenarioRand());
```

Comment:
Admittedly I was slightly taken aback by this PR. From my testing, I was under the impression that this wouldn't require any additional changes, but it clearly you've reached a different conclusion.

This change forces the future weather to be updated, which definitely gets rid of any nasty values present. I was under the impression the existing `ClimateUpdate` also took care of that, but it looks like its call to `ClimateDetermineFutureWeather` is guarded to only be called every few ticks.

Conversely, `ClimateUpdate` takes care of more business logic, such as lightning/thunder effects. However, `gameStateUpdateLogic` already calls it every tick, so I think this replacement is fine.

Reply:
The problem is that with those garbage values set it would never reach into a call to ClimateDetermineFutureWeather, also when cheats force the state it should be doing the same as when it would switch the weather which always decides the next one. This cheat just now sets the next cycle as it should have been.